services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -135,6 +136,18 @@ async def _associated_service_has_no_node_placement_contraints(
     return True
 
 
+def _by_created_dt(task: Task) -> datetime.datetime:
+    # NOTE: SAFE implementation to extract task.CreatedAt as datetime for comparison
+    if task.CreatedAt:
+        with suppress(ValueError):
+            created_at = to_datetime(task.CreatedAt)
+            created_at_utc: datetime.datetime = created_at.replace(
+                tzinfo=datetime.timezone.utc
+            )
+            return created_at_utc
+    return datetime.datetime.now(datetime.timezone.utc)
```

Comment:
Consider the following scenario:

Let's say there are two tasks, one failing and one not.
Both of Tham will raise a ValueError when parsing the date and will returns the current UTC date.
You are not guarded that the list will be sorted with first the failed and second the one not failing. Depending on how this is used can cause issues with the code.

I think this will not work for you.

Reply:
@GitHK I do not completely understand your comment.

1. if one fails and the other don't, only one (not both of them) will raise ValueError
2. if one fails, it defaults to "now" (this was in the original code although wrongly implemented)
3. the predicate do not raises 
4. this predicate is only used to sort "chronologically" pending tasks so that the oldest get attended first. 
5. I expect the failing ones to be last because typically "now" should be later that valid 
6. so the sorted list[Task] should be something like [ "old task", ..., "youngest task" ... "here an arbitrary number of failing predicates"]

If this does not answer your concern. Could you please modify the test (or create a new one) associated with this fix and reproduce the scenario you?

Reply:
OK, though the data was used for something else. All fine from my point of view.